src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -852,9 +867,22 @@ let solve_prob' resolve_ok prob logical_guard uvis wl =
              wl
     in
     commit uvis;
-    {wl with ctr=wl.ctr + 1}
-
-let extend_solution pid sol wl =
+    //record type of the uvar and its solution in the worklist
```

Comment:
cc @nikswamy , when assigning `?u <- t`, we record `(k1, k2)`, where `t:k1` and `?u:k2`, in the worklist.